[vegetablestabber] iP

vegetablestabber opened the pull request.

vegetablestabber committed:
Rename 'Duke' to 'Skeets'

vegetablestabber committed:
Allow user input

vegetablestabber committed:
Implement add and list tasks, restructure project

vegetablestabber committed:
Added the ability to mark tasks as complete or incomplete

vegetablestabber committed:
Added editor autoformatting preferences

vegetablestabber committed:
Removed unused imports

vegetablestabber committed:
Renamed parameters

vegetablestabber committed:
Delegating closing scanner to IndividualProject

vegetablestabber committed:
Set the accessibility of the `Task` constructor with the status parameters as `private`

Gavalion commented on src/main/java/Skeets.java:
```diff
@@ -0,0 +1,100 @@
+        if (!isUserExiting(input)) {
+            String command = args[0];
+
+            switch (command) {
```
great use of switches, the code is clean and you avoid arrowhead

Gavalion commented on src/main/java/Skeets.java:
```diff
@@ -0,0 +1,100 @@
+    }
+
+    private void listTasks() {
+        System.out.println(this.tasks);
```
what happens if the task is null

vegetablestabber replied:
I don't think a task can be null as all input logic is handled by the `ConsoleTools` class. However, it is possible for a task to have an empty description which I'll need to rectify. Thanks for bringing this up!

Gavalion commented on src/main/java/Skeets.java:
```diff
@@ -0,0 +1,100 @@
+        }
+    }
+
+    private boolean isUserExiting(String input) {
```
just a suggestion, if the input is empty (because of miss click or error), maybe you can ask the user "the input is empty, can you repeat what you want". Then you list down all the command available
1. list
2. mark
3. etc

vegetablestabber replied:
That's a good suggestion, thanks!

Gavalion commented on src/main/java/Skeets.java:
```diff
@@ -0,0 +1,100 @@
+        System.out.println("Task added: " + description);
+    }
+
+    private void markTask(String[] args) throws Exception {
```
great job on catching edge cases

Gavalion reviewed (commented):
great job on creating separate classes for clearer definition of the code. I do not see any coding standard issue. Keep up the good work !!!

sxkhoo commented on docs/README.md:
```diff
@@ -1,10 +1,8 @@
+# Skeets User Guide
 
-// Update the title above to match the actual product name
+![](https://static.wikia.nocookie.net/marvel_dc/images/6/65/Skeets_Prime_001.jpg/revision/latest/scale-to-width-down/1000?cb=20190703042331)
```
I like your link

sxkhoo commented on src/main/java/Skeets.java:
```diff
@@ -0,0 +1,12 @@
+    public static void main(String[] args) {
+        String logo = "     ______ ____________________\n"
+            +         "    / __/ //_/ __/ __/_  __/ __/\n"
+            +         "   _\\ \\/ ,< / _// _/  / / _\\ \\  \n"
```
I like your logo

sxkhoo commented on src/main/java/TaskList.java:
Code easily understood.

vegetablestabber committed:
Added different types of tasks

vegetablestabber committed:
Added ability to add different types of tasks

vegetablestabber committed:
Added some spacing and comments for clarity

sxkhoo commented on src/main/java/TaskList.java:
```diff
@@ -24,30 +24,30 @@ public void add(String description) {
-            action.accept(index, task);
+            successAction.accept(index, task);
         } else {
             throw new Exception("Index " + oneBasedIndex + "lies outside of valid range (1-"
```
Great use of exception for out of bounds error!

sxkhoo commented on src/main/java/ConsoleTools.java:
Code is readable and easily understood with very high quality!

vegetablestabber committed:
Removed trailing whitespaces

vegetablestabber committed:
Updated error messages for empty input

vegetablestabber committed:
Merge branch 'branch-Level-5'

vegetablestabber committed:
Fixed error message for empty input

Alfred-Goh02 commented on src/main/java/IndividualProject.java:
```diff
@@ -0,0 +1,11 @@
+
+public class IndividualProject {
+    public static void main(String[] args) {
+        Skeets skeets = new Skeets(100);
```
Avoid using magic literals and declare it as a constant at the start.

Alfred-Goh02 commented on src/main/java/Skeets.java:
```diff
@@ -0,0 +1,134 @@
+import java.util.Scanner;
+
+public class Skeets {
+
```
This file can be separated into separate classes, one for UI and one for your commands. It will be clearer for others to see the functionality that will be included in each class.

Alfred-Goh02 commented on src/main/java/ConsoleTools.java:
```diff
@@ -0,0 +1,93 @@
+    }
+
+    // Obtain the argument value for only one implicit argument
+    public static String retriveArgValue(String[] givenArgs) throws Exception {
```
Small thing but make sure the spelling is correct.

Alfred-Goh02 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+public class Deadline extends Task {
+
+    private String dueByDateTime;
+
```
The class functionality and methods are clear with the same level of abstraction good job!

Alfred-Goh02 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,34 @@
+
+    private String startDateTime;
+    private String endDateTime;
+
```
You can consider packaging all the task related classes.

Alfred-Goh02 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,33 @@
+public class Task {
+
```
Good code quality and methods are clear.

Alfred-Goh02 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,55 @@
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.stream.IntStream;
+
```
Overall very well done! For some, you can consider using native java arguments like OutOfBounds or IllegalArgument exceptions as they are more accurate. Also consider to SLAP more and separate functions into packages/classes.

vegetablestabber committed:
Implemented ability to delete tasks

vegetablestabber committed:
Updated string representation of TaskList when empty

vegetablestabber committed:
Updated gitignore

vegetablestabber committed:
Added 'raw strings' for Task classes for IO

vegetablestabber committed:
Merge branch 'branch-Level-6'

vegetablestabber committed:
Merge branch 'branch-Level-7'

vegetablestabber committed:
Restructured code into packages

vegetablestabber committed:
Exported JAR file

vegetablestabber committed:
Deleted JAR file

vegetablestabber committed:
A lot more refactoring

vegetablestabber committed:
Delegating file I/O operations to different classes

vegetablestabber committed:
Shifted Task I/O UI responsibilties to the UI class

vegetablestabber committed:
Show modified task after deletion, marking and unmarking

vegetablestabber committed:
Formatted files

vegetablestabber committed:
File reading and writing works now

vegetablestabber committed:
Ignore JAR files

vegetablestabber committed:
Removed redundant exceptions + changed date properties to LocalDate

vegetablestabber committed:
Added find functionality

vegetablestabber committed:
Added JavaDocs

vegetablestabber committed:
Merge pull request #1 from vegetablestabber/branch-Level-8

Merge Level 8 code

vegetablestabber committed:
Merge branch 'master' into branch-Level-9

vegetablestabber committed:
Merge pull request #2 from vegetablestabber/branch-Level-9

Added find functionality

vegetablestabber committed:
Merge branch 'master' into branch-A-JavaDoc

vegetablestabber committed:
Update UI.java

vegetablestabber committed:
Merge pull request #3 from vegetablestabber/branch-A-JavaDoc

Added JavaDocs

vegetablestabber committed:
Bug fixes and improvements

- Removed outdated error + updated docs for FindCommand

- Removed irrelevant throws clauses

- Packed commands for better organisation

- Removed outdated exceptions from JavaDocs

- Refactored task constructors and update input handling to improve code clarity and maintainability

- Added success message after reading tasks to enhance user feedback

- Fixed user exit condition to only check for exit command representation

vegetablestabber committed:
Further edits to .md files for user-friendly documentation

- Added initial draft for docs/README.md
- Added alt text for Skeets image
- Updated README.md
- Added user-friendly instructions on how to use the bot
- Split the methods into sections for better readability

vegetablestabber committed:
Changed how tasks were written on exiting the program since the previous process was not working

vegetablestabber committed:
Add documentation to exit the program
